planner/core/find_best_task.go

```diff
@@ -582,6 +582,22 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 			candidates = append(candidates, currentCandidate)
 		}
 	}
+
+	if ds.ctx.GetSessionVars().GetAllowAlwaysPreferIndex() {
+		noTablePathCandidate := make([]*candidatePath, 0, 4)
+		tablePathCandidate := make([]*candidatePath, 0, 4)
+		for _, c := range candidates {
+			if !c.path.IsTablePath() {
```

Comment:
There should be no more than one table path. We can just find it and remove it when the path is more than one

Reply:
Have been modified. 😊